apollo-federation/src/schema/type_and_directive_specification.rs

```diff
@@ -440,44 +459,11 @@ impl DirectiveSpecification {
                     arg_strategies.len() == args.len(),
                     "Invalid directive specification for @{name}: not all arguments define a composition strategy"
                 );
-                let name_capture = name.clone();
-                let args_capture = args.to_vec();
-                argument_merger = Some(Box::new(move |schema: &FederationSchema| -> Result<ArgumentMerger, SingleFederationError> {
-                    for arg in args_capture.iter() {
-                        let strategy = arg.composition_strategy.as_ref().unwrap();
-                        let arg_name = &arg.base_spec.name;
-                        let arg_type = (arg.base_spec.get_type)(schema)?;
-                        assert!(!arg_type.is_list(), "Should have gotten error getting type for @{name_capture}({arg_name}:), but got {arg_type}");
-                        strategy.is_type_supported(schema, &arg_type).map_err(|support_msg| {
-                            let strategy_name = strategy.name();
-                            SingleFederationError::DirectiveDefinitionInvalid {
-                                message: format!("Invalid composition strategy {strategy_name} for argument @{name_capture}({arg_name}:) of type {arg_type}; {strategy_name} only supports ${support_msg}")
-                            }
-                        })?;
-                    }
-                    let arg_strategies_capture = arg_strategies.clone();
-                    let arg_strategies_capture2 = arg_strategies.clone();
-                    Ok(ArgumentMerger {
-                        merge: Box::new(move |arg_name: &str, values: &[Value]| {
-                            let Some(strategy) = arg_strategies_capture.get(arg_name) else {
-                                panic!("`Should have a strategy for {arg_name}")
-                            };
-                            strategy.merge_values(values)
-                        }),
-                        to_string: Box::new(move || {
-                            if arg_strategies_capture2.is_empty() {
-                                "<none>".to_string()
-                            }
-                            else {
-                                let arg_strategy_strings: Vec<String> = arg_strategies_capture2
-                                    .iter()
-                                    .map(|(arg_name, strategy)| format!("{arg_name}: {}", strategy.name()))
-                                    .collect();
-                                format!("{{ {} }}", arg_strategy_strings.join(", "))
-                            }
-                        }),
-                    })
-                }));
+                argument_merger = Some(directive_argument_merger(
```

Comment:
This is just a pure refactoring. `cargo fmt` was adding another indentation level, so I factored out the whole block.